### DISCUSSION TITLE
Get the geometries of the edges for a given area

### DISCUSSION THREAD

**Discussion by Bondify** [valhalla/valhalla#3909] [Q&A] [answered]
I'm hosting my own Valhalla server and calling it's APIs for a variety of things (it's great, thanks for working on this!).
I would need to geometry of the street network in order to create a few variables for each street edge. Things like number of neighbors, area, etc. For most of the variables I need the geometry of the edges in the area I'm studying.

Do I have a way of reading the tiles of a given area? I see the folder where they are but haven't found a way to read them and get the geometry and I don't see it on the docs.

Or maybe I can get it from a different python library? The thing is that I'm already assigning a few variables to each edge_id so if I use a different source I might not get the same edge_ids. 

Thanks for your help!

**Comment by nilsnolde** [comment, marked as answer]
Either [`/locate`](https://valhalla.github.io/valhalla/api/locate/api-reference/) or [`/trace_attributes`](https://valhalla.github.io/valhalla/api/map-matching/api-reference/#trace-attributes-action)? You'll need only the lib for converting the decoded polyline to your preferred format for LineStrings.

There's currently no way to get the geometries covered by e.g. a polygon, i.e. by area.

**Reply by Bondify** [reply]
Thanks @nilsnolde. Do you know if the edge_ids used by Valhalla match the ones of OSM if I just downloaded it the graph from OSM?

**Reply by nilsnolde** [reply]
Edge id (Valhalla) != OSM ID, so no, but you can get access to the OSM ID through the endpoints I listed
